qt/scientific_interfaces/Inelastic/QENSFitting/FitPlotView.cpp

```diff
@@ -15,6 +15,7 @@
 #include <QSignalBlocker>
 #include <QTimer>
 
+#include <gtest/gtest-param-test.h>
```

Comment:
Same question here. I am also interested, is there a reason why we sometimes use lambdas to connect a slot?

Reply:
The use of static_cast is for overloaded signals or slots. It's the way recommended by qt: https://wiki.qt.io/New_Signal_Slot_Syntax . 
We could use the class qoverloaded for this, but it's only available from qt 5.7, which we don't use yet. 

The lambdas I use for 2 situations: cppcheck gave an error that the signal `currentIndexChanged(Qstring)`  emitted from a  QComboBox is deprecated and `currentIndexChanged(int)` must be used. I use the lambda to avoid refactoring the slot to take the `int` argument, so the text from the index is retrieved in the lambda function directly. 
The other case for lambdas was when there was a slot with a default argument. Otherwise it doesn't capture correctly the default argument.

Reply:
Ah I see, thanks for the explanation! When you get the chance, can you open a maintenance issue to switch from using `currentIndexChanged(QString)` to `currentIndexChanged(int)`?